frontend/apps/reader/modules/readerrolling.lua

```diff
@@ -1170,6 +1181,7 @@ function ReaderRolling:showEngineProgress(percent)
         -- No need for any paint/refresh: any action we got
         -- some progress callback for will generate a full
         -- screen refresh.
+     self.progress_last_y_shift = self.ui.document:getHeaderHeight()
```

Comment:
Not super sure this is fine: you save it here, so it's what will be used on next re-renderings.
But depending on stuff, this progress (which is shown only after a delay, so it's not when rendering are fast, less than a few seconds) might not be shown, but could be on a next re-rendering (say, huge style tweak increasing rendering time), and this progress_last_y_shift would not have been set.

I'd rather try with your initial simple idea and my https://github.com/koreader/koreader/pull/7667#issuecomment-839930085, so the state is store in the current widget.